### PR TITLE
feat: B2B Node SDK

### PR DESCRIPTION
**Pull request by max-stytch**
- Scaffolds out the Node Stytch B2B SDK
- Implements the following B2B modules:
    - Members
    - Organizations
    - SSO
    - SAML
    - Member Sessions
    - Magic Links
- Refactors the B2B and B2C SDKs to use a shared base client

### CODE REVIEW

**Review by anmaini-ucsd — APPROVED** [review]
Nit in `LoginOrSignupByEmailResponse` but other than that LGTM!

File: lib/b2b/magic_links.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  signup_redirect_url?: string;
+  pkce_code_challenge?: string;
+  login_template_id?: string;
+  signup_template_id?: string;
+}
+
+export interface LoginOrSignupByEmailResponse extends BaseResponse {
+  member_id: string;
+  member: Member;
+  member_creared: boolean;
```
Comment (anmol-stytch) [inline lib/b2b/magic_links.ts:17]: nit: `member_created`

**Review by anmol-stytch — APPROVED** [review]
Nit in LoginOrSignupByEmailResponse but other than that it looks good to me

**Review by taronish-stytch — APPROVED** [review]
lgtm

File: lib/b2b/client.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  magicLinks: MagicLinks;
+  sessions: Sessions;
+  organizations: Organizations;
+  sso: SSO;
+
+  constructor(config: ClientConfig) {
+    super(config);
+
+    if (!this.fetchConfig.baseURL.endsWith("b2b/")) {
+      this.fetchConfig.baseURL += "b2b/";
```
[the remark this answers is not in the document]
Comment (max-stytch) [reply lib/b2b/client.ts:17]: Our base URL is our env string - `stytch.envs.live` or `stytch.envs.test` - I didn't want to introduce a separate set of envs for B2B, I think that would make things more confusing for end users / harder to document. One other option would be to embed the `b2b/` URL in all the b2b-scoped client endpoints

**Review by danny-stytch — APPROVED** [review]
LGMT

**Pull request merged**